scripts/dasharo-deploy

```diff
@@ -367,87 +628,224 @@ blob_transmission() {
   if [ -n "$SCH5545_FW" ]; then
     error_file_check "$SCH5545_FW" "Failed to find SCH5545 EC firmware binary."
     echo -n "Adding SCH5545 EC firmware..."
-    cbfstool "$BIOS_UPDATE_FILE" add -f "$SCH5545_FW" -n sch5545_ecfw.bin -t raw
+    $CBFSTOOL "$BIOS_UPDATE_FILE" add -f "$SCH5545_FW" -n sch5545_ecfw.bin -t raw
     print_ok "Done"
   fi
 
   if [ -n "$ACM_BIN" ]; then
     error_file_check "$ACM_BIN" "Failed to find BIOS ACM binary."
     echo -n "Adding BIOS ACM..."
-    cbfstool "$BIOS_UPDATE_FILE" add -f "$ACM_BIN" -n txt_bios_acm.bin -t raw
+    $CBFSTOOL "$BIOS_UPDATE_FILE" add -f "$ACM_BIN" -n txt_bios_acm.bin -t raw
     print_ok "Done"
   fi
 
   if [ -n "$SINIT_ACM" ]; then
     error_file_check "$SINIT_ACM" "Failed to find Intel SINIT ACM binary."
     echo -n "Adding SINIT ACM..."
-    cbfstool "$BIOS_UPDATE_FILE" add -f "$SINIT_ACM" -n txt_sinit_acm.bin -t raw
+    $CBFSTOOL "$BIOS_UPDATE_FILE" add -f "$SINIT_ACM" -n txt_sinit_acm.bin -t raw
     print_ok "Done"
   fi
 }
 
-install_ec() {
-  verify_artifacts ec
-  echo "Installing EC..."
-  $FLASHROM -p "$PROGRAMMER_EC" ${FLASH_CHIP_SELECT} -w "$EC_UPDATE_FILE" >> $FLASHROM_LOG_FILE 2>> $ERR_LOG_FILE
-  error_check "Failed to install Dasharo EC firmware"
-  echo "Successfully installed Dasharo EC firmware"
-}
+deploy_ec_firmware() {
+# This function deploys (installs or updates) downloaded EC firmware either UEFI
+# capsules (updates only) and binaries. Parameters: update, install.
+#
+# TODO: Currently we have here flashrom parameters configuration code, this
+# should be done before this function is called, so as to place here only
+# deployment-related code. Ideally the deploying calls would look like this:
+#
+# $DEPLOY_COMMAND $DEPLOY_ARGS &>> $LOGS_FILE
+  local _mode
+  _mode="$1"
+
+  if [ "$_mode" == "update" ]; then
+    echo "Updating EC..."
+
+    # The EC firmware could be updated in two ways: via UEFI Capsule Update or
+    # via binaries and flashrom:
+    if [ "$FIRMWARE_VERSION" == "community_cap" ] || [ "$FIRMWARE_VERSION" == "dpp_cap" ]; then
+      # Linux Kernel driver is responsible for handling UEFI Capsule Updates, so
+      # the capsule should be fed to a specific device:
+      $CAP_UPD_TOOL "$EC_UPDATE_FILE"
```

Comment:
Maybe you should restart after this to keep it consistent with the way `DASHARO_ECTOOL flash` works and `# Ends in a reset, does not exit.` comment?

Reply:
From internal discussion with dasharo-team I got info, that every EC update (does not matter whether it is capsule or not) causes a restart of EC, and, therefore, the restart of the entire system. And even more, I got info, that EC firmware may reside in one capsule with main Coreboot firmware, therefore even `deploy_firmware update` might cause a reboot.

Reply:
This is wrong interpretation of what was said. `deploy_firmware update` by itself won't cause a reboot, because the capsule isn't applied just by writing to that device. It is firmware that parses and applies the capsule on next reboot, which must be initiated by DTS.

Reply:
Good to know, thank you :).

Reply:
> and # Ends in a reset, does not exit. comment?

I [have deleted](https://github.com/Dasharo/dts-scripts/compare/9aa00a04166f712f638acc875e4004c28ece5c32..64340aab680221575d03a727f1e986ce4ace85dd) the comment.

> Maybe you should restart after this to keep it consistent with the way

I do not see a reason to do so, just "to keep consisted"  is not a reason IMHO.

Reply:
` is not a reason IMHO`

Really? Currently platform is always rebooted after update except if you do cap EC update.

Reply:
Not only that but above `deploy_ec_firmware update` we have:

```sh
echo "Updating Embedded Controller firmware. Your computer will power off automatically when done."
```

which also isn't true right now.

Reply:
I see I have missed reboot condition, fixed here e4eaaa8808290c9a6dd2640a21f850e8156fad0d.